api/migrations/20200603153526-bcmiDataLoad.js

```diff
@@ -0,0 +1,146 @@
+
+    // API call to pull data from BCMI
+    // could also fetch from NRPTI first: require('../src/models/bcmi/mine').find().then(...);
+    // then match to BCMI, just doing it this way so we can match on name, rather then code
+    console.log('Fetching all major mines in BCMI...');
+    // 30 results for major, 74 for published. Note, published does not return links so if we're supposed to use
+    // published, we'll need to do a follow up call to /api/project/bycode/<mineData.code> to get the links
+    const publishedMines = await getRequest(bcmiUrl + '/api/projects/published');
+    console.log('Located ' + publishedMines.length + ' mines. Batching updates...');
+
+    const promises = [];
+    // build up a collection of all requests
+    for (let i = 0; i < publishedMines.length; i++) {
+      try {
+        // The published endpoint doesn't have links, refresh the mineData object
+        let mineData = await getRequest(bcmiUrl + '/api/project/bycode/' + publishedMines[i].code);
+        promises.push(updateMine(mineData, nrpti));
+      } catch(err) {
+        console.error('Could not find ' + publishedMines[i]._id + ' : ' + publishedMines[i].name);
+        // dont rethrow, we'll just ignore this one as a failure and check the rest
+      }
+    }
+
+    // fire off the requests and wait
+    let results = await Promise.all(promises);
+
+    let updatedCount = 0;
+    let notFoundCount = 0;
+    results.forEach(result => {
+      if (Object.prototype.hasOwnProperty.call(result, 'notfound')) {
+        notFoundCount++;
+        console.log('Could not find ' + result.data._id + ' : ' + result.data.name);
+      } else {
+        updatedCount++;
+      }
+    });
+
+    // we're done
+    console.log('BCMI migration complete.');
+    console.log('Of ' + publishedMines.length + ' mines in BCMI, ' + notFoundCount + ' could not be found in NRPTI, and ' + updatedCount + ' were updated.');
+  } catch(err) {
+    console.error('Error on BCMI dataload: ' + err);
+  }
+
+  mClient.close();
+};
+
+exports.down = function(db) {
+  return null;
+};
+
+exports._meta = {
+  "version": 1
+};
+
+async function updateMine(mineData, nrpti) {
+  let nrptiMines = await nrpti.find({ _schemaName: 'MineBCMI', name: mineData.name}).toArray();
+  // should have 1 result returned. Any more or less, just ignore this update
+  if (nrptiMines.length === 1) {
+    let externalLinks = [];
+    // format the links from the data
+    for(const idx in mineData.externalLinks) {
+      const link = mineData.externalLinks[idx];
+      externalLinks.push(link.link);
+    }
+
+    return nrpti.update({ _id: new ObjectID(nrptiMines[0]._id) }, {
+      $set: {
+        type:            mineData.type,
+        summary:         mineData.description, // BCMI doesn't have a "summary" attribute
+        description:     mineData.description,
+        links:           externalLinks,
+        updatedBy:       'NRPTI BCMI Data Migration',
+        sourceSystemRef: 'mem-admin'
```

Comment:
I don't think we want to change the sourceSystemRef here.  

In this case it will have been set to 'core' by the core importer.  But for any of these records, it should reflect whichever 'source' system it was imported from, and which it will receive periodic updates from.

Reply:
I disagree - mem-admin is the sourceSystemRef since this is where the data is coming from.  It's not coming from core, but from mem-admin service: https://mines.empr.gov.bc.ca/

Reply:
Maybe I am misunderstanding the purpose of this migration.  The code indicates that it is only updating an existing mine record.  Where does the original mine record come from? I assumed it was Core, based on the ticket description, but maybe that isn't the case.

Reply:
It's just what data source endpoint this came from.  We say epic because we're calling epic api and that's where we pulled that data from, same for nris-epd.  We also store refId's based on the called system, as we don't really know if that system created it or pulled it in from somewhere else.

Reply:
I guess this is mainly what I'm wondering:

If we run the core importer and create a new record from core, and then update it via migration from bcmi, and then update it again in future by running the core importer again, would the sourceSystemRef change each time (core -> mem-admin -> core) or would it stay fixed as core?